### PR TITLE
TS: Make subclass inheritance more comfortable

### PR DESCRIPTION
**Pull request by eos3tion**
Make subclass inheritance more comfortable
remove same `copy` function

### CODE REVIEW

File: src/core/Object3D.d.ts [hunk 1]
```diff
@@ -346,6 +346,6 @@ export class Object3D extends EventDispatcher {
 	 * @param object
 	 * @param recursive
 	 */
-	copy( source: Object3D, recursive?: boolean ): this;
+	copy( source: this, recursive?: boolean ): this;
```
Comment (Mugen87) [inline src/core/Object3D.d.ts]: I'm not sure `this` is correct in this context. `this` refers always to the object where the method invocation was performed. It seems to me it sense to use it when defining the return value/type but not in context of `copy()`s parameters.

Comment (gkjohnson) [reply src/core/Object3D.d.ts]: I was a little confused by this as well but when testing in VSCode it looks like `this` refers to the type of the current object when used in a function parameter. I was having trouble find typescript documentation that explains this, though.

Comment (Mugen87) [reply src/core/Object3D.d.ts]: Okay, thanks for the explanation. In this case, the change is fine.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**